### PR TITLE
Refs #30648 - Allow overriding proxy selector at the feature level

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

**Comment by adamruzicka** [comment]
This is about a RemoteExecutionFeature model in Foreman, not a proxy feature. Or do you mean using the new capabilities api on the smart proxy to determine which one is the "internal" one?

**Comment by ezr-ondrej** [comment]
> This is about a RemoteExecutionFeature model in Foreman, not a proxy feature. Or do you mean using the new capabilities api on the smart proxy to determine which one is the "internal" one?

That is exactly what I mean :)
We are trying to run this exact feature on internal proxy only.
But instead we can ask the proxy what features it supports and only internal would answer it supports this feature :shrug:

**Comment by adamruzicka** [comment]
We were considering going that route but because we started on this pretty late, we decided to go the simpler, but arguably less correct route, see discussion at https://github.com/theforeman/foreman_remote_execution/pull/490

**Comment by ezr-ondrej** [comment]
Oh, so we want to backport this?

File: db/migrate/20200820122057_add_proxy_selector_override_to_remote_execution_feature.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddProxySelectorOverrideToRemoteExecutionFeature < ActiveRecord::Migration[4.2]
+  def change
+    add_column :remote_execution_features, :proxy_selector_override, :string
```
Comment (ezr-ondrej) [inline db/migrate/20200820122057_add_proxy_selector_override_to_remote_execution_feature.rb:3]: Would `proxy_selector` do?
```suggestion
    add_column :remote_execution_features, :proxy_selector, :string
```
Apart of that, lets get it in! :+1:

**Review by ezr-ondrej — APPROVED** [review]

File: db/migrate/20200820122057_add_proxy_selector_override_to_remote_execution_feature.rb (hunk 1 above)
Comment (adamruzicka) [reply db/migrate/20200820122057_add_proxy_selector_override_to_remote_execution_feature.rb:3]: With the `_override` suffix it is clear that if set it overrides some other value and if unset, some "normal" value is used. If we change it to `proxy_selector` only I'd say it stops being clear what no value in this column means

Comment (ezr-ondrej) [reply db/migrate/20200820122057_add_proxy_selector_override_to_remote_execution_feature.rb:3]: Fair enough. I still think blank value = I do not care 'bout my proxy is fair as well, but ok. No strict opinion here.

**Comment by adamruzicka** [comment]
Needs https://github.com/theforeman/foreman_remote_execution/pull/536 to prevent issues where features are registered before db is migrated

**Pull request merged**